tests/ImageSharp.Tests/TestFile.cs

```diff
@@ -65,7 +66,28 @@ public sealed class TestFile
         /// <summary>
         /// Gets the image with lazy initialization.
         /// </summary>
-        private Image<Rgba32> Image => this.image ??= ImageSharp.Image.Load<Rgba32>(this.Bytes);
+        private Image<Rgba32> Image
+        {
+            get
+            {
+                Image<Rgba32> img = this.image;
+                if (img is null)
+                {
+                    Image<Rgba32> loadedImg = ImageSharp.Image.Load<Rgba32>(this.Bytes);
```

Comment:
Well, we can do a double check lock instead so image would never be loaded twice:
```cs
// this.image should be marked volatile still
if (this.image is null)
{
    lock(this.lockObject)
    {
        if (this.image is null)
        {
            this.image = ImageSharp.Image.Load<Rgba32>(this.Bytes);
        }
    }
}
return this.image;
```

I proposed `Interlocked` solution, I know, but `lock` solution is a lot more readable IMO and doesn't have a possible flaw of loading the same image twice. It's also isn't slow on reads due to locking as most of the time first `if` check would be true.

Maybe we should use it instead due to simplicity?

Reply:
Definitely far more readable. I’m happy to switch out